price_rule.go

```diff
@@ -84,6 +84,73 @@ type PriceRulesResource struct {
 	PriceRules []PriceRule `json:"price_rules"`
 }
 
+// SetPrerequisiteSubtotalRange sets or clears the subtotal range for which a cart must lie within to qualify for the price-rule
+func (pr *PriceRule) SetPrerequisiteSubtotalRange(greaterThanOrEqualTo *string) error {
+	if greaterThanOrEqualTo == nil {
+		pr.PrerequisiteSubtotalRange = nil
+	} else {
+		if !validateMoney(*greaterThanOrEqualTo) {
+			return fmt.Errorf("failed to parse value as Decimal, invalid value")
+		}
+
+		pr.PrerequisiteSubtotalRange = &prerequisiteSubtotalRange{
+			GreaterThanOrEqualTo: *greaterThanOrEqualTo,
+		}
+	}
+
+	return nil
+}
+
+// SetPrerequisiteQuantityRange sets or clears the quantity range for which a cart must lie within to qualify for the price-rule
+func (pr *PriceRule) SetPrerequisiteQuantityRange(greaterThanOrEqualTo *int) {
+	if greaterThanOrEqualTo == nil {
+		pr.PrerequisiteQuantityRange = nil
+	} else {
+		pr.PrerequisiteQuantityRange = &prerequisiteQuantityRange{
+			GreaterThanOrEqualTo: *greaterThanOrEqualTo,
+		}
+	}
+}
+
+// SetPrerequisiteShippingPriceRange sets or clears the shipping price range for which a cart must lie within to qualify for the price-rule
+func (pr *PriceRule) SetPrerequisiteShippingPriceRange(lessThanOrEqualTo *string) error {
+	if lessThanOrEqualTo == nil {
+		pr.PrerequisiteShippingPriceRange = nil
+	} else {
+		if !validateMoney(*lessThanOrEqualTo) {
+			return fmt.Errorf("failed to parse value as Decimal, invalid value")
+		}
+
+		pr.PrerequisiteShippingPriceRange = &prerequisiteShippingPriceRange{
+			LessThanOrEqualTo: *lessThanOrEqualTo,
+		}
+	}
+
+	return nil
+}
+
+// SetPrerequisiteToEntitlementQuantityRatio sets or clears the ratio between ordered items and entitled items (eg. buy X, get y free) for which a cart is eligible in the price-rule
+func (pr *PriceRule) SetPrerequisiteToEntitlementQuantityRatio(prerequisiteQuantity *int, entitledQuantity *int) {
```

Comment:
There are quite a few restrictions on the use of these Prerequisites and specifically this one. 
![image](https://user-images.githubusercontent.com/1119730/108379131-3ff12480-71cb-11eb-803e-4f0709797534.png) Should we add more documentation or checks on this or is it reasonable to assume that the user will have read the Shopify documentation and understand those restrictions. I haven't use this API myself if one of these restrictions is broken I'm curious if enough information is returned for the user to diagnose the issue.

Reply:
I did a little testing and it appears that the messaging around these constraints is actually quite good:
![image](https://user-images.githubusercontent.com/73915353/108384905-c5c39e80-71d0-11eb-9a88-6d5a0ba28dcf.png)

I think it is reasonable to assume that the user will read the platform documentation, but even if they did not, I feel that this is verbose enough. In addition I'm a little leery to add very tight validation on this library, as it sits in front of a platform that we do not control. We would be making goshopify potentially quite brittle to changes of that platform validation, as such I tend to prefer letting the platform perform its own validation and allowing consumers of this library to instead impose their own constraints as needed.

Based on this example, what do you think?

Reply:
Works for me, thanks for looking into this and I agree the extra validation is a rabbit hole I'd want to avoid if possible.